_docs/versioning.md

```diff
@@ -0,0 +1,184 @@
+    - main
+    - v3.2
+    - beta
+    - v3.1
+```
+
+## Table of Contents Handling (TOC)
+
+As with the documents, your TOC contents also need to be updated so they correctly point to the document versions you are viewing.
+
+You will need to update the toc file as the structure of your documentation/toc changes over time but as we have taken a snapshot of our documentation we need to do the same thing for our toc so it reflects the document structure at that point in time too. You will see that for this site we have the versions of Current and Previous and you will see that the Current version has all the information about versioning which didn't exist in the prior version.
+
+To enable the TOC to change based on the version, we have to create a new toc file and name it {version}-toc.yml, so in this site we have created previous-toc.yml in the _data directory that has the TOC structure for the prior version where we did not have the versioning.md file present. 
+
+
+{% include alert.html type="info" title="Note: The toc.yml file will always be our current table of contents pointing to the most recent version of the documentation in the base of our _docs folder. When adding additional items to your toc.yml file you do not need to worry about the version identifier as that is taken care of by Docsy Jekyll when viewing alternative versions, continue to have the urls pointing to the files in the base docs directory." %}
+
+Here is what our current toc.yml file in our _data directory looks like, the previous-toc.yml file will look identical apart from omitting the 'Versioning' title and link which was added for this release.
+
+```yml
+title: Documentation
+  url: docs
+  links:
+    - title: "Getting Started"
+      url: "docs/getting-started"
+      children:
+        - title: Features
+          url: "docs/getting-started#getting-started"
+        - title: Development
+          url: "docs/getting-started#development"
+        - title: Customization
+          url: "docs/getting-started#customization"
+    - title: Versioning
+      url: "docs/versioning"
+    - title: "About"
+      url: "about"
+    - title: "News"
+      url: "news"
+- title: "Extras"
+  url: "docs/extras"
+  links:
+    - title: Quizzes
+      url: "docs/extras/example-quiz"
+    - title: Tags Page
+      url: "tags"
+```
+
+Now not every version of your documentation will have a change in structure and therefore require an update to toc.yml. To remove this overhead a mapping file is used to tell Docsy Jekyll which TOC it should use for a particular version,
+the benefit of this is that if the structure of your site does not change then you do not need to create an additional toc.yml file.
+
+You will have a toc-mapping.yml file in your _data directory and the contents will contain the code below to map a particular version with a particular TOC file to use. You need to update this file for each version you create even if you do not need to create a new TOC file.
```

Comment:
For my note above - I think this mapping would be fine to leave at the top level, and (per the same reason of many versions) we should have a subfolder for the archived ones.

Reply:
I did leave the mapping file at the top level and made the versions available in a user configurable subfolder.